src/main/kotlin/io/github/mojira/arisa/ArisaMain.kt

```diff
@@ -1,165 +1,68 @@
 package io.github.mojira.arisa
 
-import arrow.syntax.function.partially1
 import ch.qos.logback.classic.AsyncAppender
 import ch.qos.logback.classic.LoggerContext
 import com.github.napstr.logback.DiscordAppender
 import com.uchuhimo.konf.Config
 import com.uchuhimo.konf.source.yaml
-import io.github.mojira.arisa.domain.Issue
-import io.github.mojira.arisa.infrastructure.HelperMessageService
-import io.github.mojira.arisa.infrastructure.ProjectCache
 import io.github.mojira.arisa.infrastructure.config.Arisa
-import io.github.mojira.arisa.infrastructure.jira.connectToJira
-import io.github.mojira.arisa.infrastructure.jira.toDomain
-import io.github.mojira.arisa.registry.TicketQueryTimeframe
-import io.github.mojira.arisa.registry.getModuleRegistries
 import net.rcarz.jiraclient.JiraClient
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
-import java.io.File
-import java.lang.Long.max
-import java.time.Duration
-import java.time.Instant
-import java.time.temporal.ChronoUnit
 import java.util.concurrent.TimeUnit
 
 val log: Logger = LoggerFactory.getLogger("Arisa")
 
-private const val TIME_MINUTES = 5L
-private const val MAX_QUERY_TIMEFRAME_IN_MINUTES = 10L
-
-const val MAX_RESULTS = 50
+lateinit var jiraClient: JiraClient
 
-private const val RELOG_INTERVAL_IN_SECONDS = 5 * 60L
-private const val WAIT_TIME_AFTER_CONNECTION_ERROR_IN_SECONDS = 40L
+private val config = readConfig()
+private val connectionService = JiraConnectionService(config)
+private var executor = Executor(config)
```

Comment:
Yes it can! Previously the executor was constantly changed from the outside, I've changed it so that it now always accesses the global `jiraClient` instead of having a separate copy (which is not very nice either but it works and we don't need to allocate a new executor on every relog).